### PR TITLE
test(integration): add integration test for master lost during syncing sst files.

### LINKED ISSUE #2671: Add more test cases about replication and cluster
By the way, can we add a test with master removed and about 5s' connection timeout in go integration?

_Originally posted by @mapleFU in https://github.com/apache/kvrocks/issues/2662#issuecomment-2477964921_

### PR DESCRIPTION
**Pull request by LindaSummer**
# Issue

Close #2671 

# Proposed Changes

Add integration test for losing master during SST file syncing.

Here is the steps.

1. Create master and slave instances.
2. Apply master and slave setnodeid.
3. Create a slow tcp proxy for master.
4. Setnodes to master and write data to master.
5. Setnodes to slave.
6. Wait for one second and close tcp proxy.
7. Try setnodes in slave and get duration.

### CODE REVIEW

**Comment by LindaSummer** [comment]
Hi,

I'm not sure my case is correctly covered the scenario in related issue.

I only got about 1 second delay in my own machine.

Please let me know if there is any problem about the test case. 😊

Best Regards,
Edward

**Review by PragmaTwice — APPROVED** [review]
Thank you for your contribution! See if some others have comments.

File: tests/gocase/integration/replication/replication_test.go [hunk 1]
```diff
@@ -552,3 +552,60 @@ func TestFullSyncReplication(t *testing.T) {
...
+	}
+
+	require.NoError(t, replicaClient.Do(ctx, "clusterx", "SETNODES", clusterNodesInfo, "1").Err())
+
+	time.Sleep(2 * time.Second)
+	cancelProxy()
+	start := time.Now()
+	require.NoError(t, replicaClient.Do(ctx, "clusterx", "SETNODES", unexistNodesInfo, "2").Err())
+	duration := time.Since(start)
+	require.Less(t, duration, time.Second*6)
```
Comment (mapleFU) [inline tests/gocase/integration/replication/replication_test.go:610]: Why the time out is 5000/5100 but the duration be 6 here 🤔

**Review by git-hulk — APPROVED** [review]
LGTM, great work.

File: tests/gocase/integration/replication/replication_test.go (hunk 1 above)
Comment (LindaSummer) [reply tests/gocase/integration/replication/replication_test.go:610]: Hi @mapleFU ,

The time duration may not be a precise value due to network.

So I reserved 1 second to reduce flaky failure. 😊

Best Regards,
Edward

Comment (mapleFU) [reply tests/gocase/integration/replication/replication_test.go:610]: What would a duration really looks like? Would it:
1. Force greater than 5?
2. Always less than 6? (I guess this is not guranteed ...)

Comment (LindaSummer) [reply tests/gocase/integration/replication/replication_test.go:610]: Hi @mapleFU ,

Yes, I thought it should be in a [5, 6 or longer] range.

But I only got about one second in my own environment and make me confused and not sure whether the case full covers the issue. 🤔

Maybe we can set the upper bound to a conservative value just for integration test. 😊

Best Regards,
Edward

Comment (mapleFU) [reply tests/gocase/integration/replication/replication_test.go:610]: > But I only got about one second in my own environment and make me confused and not sure whether the case full covers the issue. 🤔

Thanks! I'll dive in my env later, we can first move forward

**Pull request merged**